### ISSUE TITLE
Cannot start language server:  Message: URI "" is not valid

### ISSUE THREAD

**Issue by oising** [hashicorp/terraform-ls#158]
### Server Version

```
0.3.2
```

### Terraform Version

```
Terraform v0.12.26
```

### Client Version

```
VSCode 1.46.0
```

### Terraform Configuration Files

```hcl
provider "azurerm" {
  version = "~> 2.0"
  features { }
}
```

### Log Output

```text
Launching language server: c:\Users\OisinGrehan\.vscode\extensions\hashicorp.terraform-2.0.1/lsp/terraform-ls serve
2020/06/11 15:59:16 service.go:63: Preparing new session ...
2020/06/11 15:59:16 langserver.go:79: Starting server (pid 14976) ...
2020/06/11 15:59:16 server.go:469: Received 1 new requests
2020/06/11 15:59:16 server.go:165: Processing 1 requests
2020/06/11 15:59:16 server.go:223: Checking request for "initialize": {"processId":38236,"rootPath":null,"rootUri":null,"capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional"},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true},"textDocument":{"publishDiagnostics":{"relatedInformation":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true}}},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true},"codeAction":{"dynamicRegistration":true,"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true},"documentLink":{"dynamicRegistration":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true}}},"trace":"off","workspaceFolders":null}
2020/06/11 15:59:16 rpc_logger.go:29: Incoming request for "initialize" (ID 0): {"processId":38236,"rootPath":null,"rootUri":null,"capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional"},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true},"textDocument":{"publishDiagnostics":{"relatedInformation":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true}}},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true},"codeAction":{"dynamicRegistration":true,"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true},"documentLink":{"dynamicRegistration":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true}}},"trace":"off","workspaceFolders":null}
2020/06/11 15:59:16 rpc_logger.go:45: Error for "initialize" (ID 0): [-32098] URI "" is not valid
2020/06/11 15:59:16 server.go:204: Completed 1 requests [14.354ms elapsed]
[Error - 3:59:16 p.m.] Server initialization failed.
  Message: URI "" is not valid
  Code: -32098 
```

### Expected Behavior

Language server should start.
### Actual Behavior

Language server fails to start.
### Steps to Reproduce

f1, "enable terraform language serve"

**Comment by radeksimko** [comment]
For context the single file support is tracked under https://github.com/hashicorp/terraform-ls/issues/32 but I will keep this issue open until we fix the confusing messaging around this.

**Comment by oising** [comment]
I opened VS, created new file, saved it as .tf then tried to start the LS. @radeksimko - you're saying it will work if I open a folder instead of a single file?

**Comment by oising** [comment]
ok, now it started but the suggestions are all doubled up -- I'll open a new issue for this.

**Comment by radeksimko** [comment]
@oising Generally the expectation is that only the simplest scenario currently works, which can be described as

 - having a folder, which was `terraform init`d prior to opening it in VSCode (or any other IDE)
 - the same folder is opened as "root" in VSCode (or any other IDE)
 - the folder does not contain any subfolder, or you don't open files in these subfolders

https://github.com/hashicorp/terraform-ls/issues/32 lists a number of scenarios which don't work yet (but will eventually). Vast majority of the ones we don't support yet is surfaced as "Workspace not initialized". That message can be misleading in many valid scenarios, but that is the current state of things.

**Comment by techadmin1982** [comment]
why was this case closed ? has this problem resolved ?

**Comment by dguymon** [comment]
I was able to resolve the same issue on my end by ensuring that my $PATH environment variable in my ~/.bash_profile file was fully-qualified (i.e. no ~ in the URI), that I opened the top-level folder container all the Terraform code within the repository in VSCode, and ensuring I had previously run `terraform init` in the project before doing so to suppress the various language server errors from the VSCode Terraform plugin during VSCode launch.

**Comment by radeksimko** [comment]
Editing single files will be reported correctly from the next release (probably 0.4.0; to be released in coming days) as _"Editing a single file is not yet supported. Please open a directory."_. That is why this was closed as part of #172 (which the closing event mentions).

Supporting single files as something yet to be solved as part of https://github.com/hashicorp/terraform-ls/issues/32 
Please follow that issue if you are interested in knowing when we support single files.
